File: CocoatainerGuiExample/CocoatainerGuiExample/GuiCocoaMug/Marshmallow.m
//
//  Marshmallow.m
//  CocoatainerExample
//
//  Created by Jeffrey Bakker on 2015-05-14.
//  Copyright (c) 2015 Jeffrey Bakker. All rights reserved.
//

#import "Marshmallow.h"

@interface Marshmallow ()
{
@private
    id<ILog> _log;
}
@end

@implementation Marshmallow

-(id)initWithLog:(id<ILog>)log
{
    self = [super init];
    if (self)
    {
        _log = log;
    }
    return self;
}

-(void)dealloc
{
    [_log write:
     @"This marshmallow is so soggy that it has nearly turned into liquid."];
}

-(NSString*)name
{
    return NSStringFromClass([self class]);
}

@end

File: CocoatainerSwiftExample/CocoatainerSwiftExample/SwiftCocoaMug/Marshmallow.swift
//
//  Marshmallow.swift
//  CocoatainerSwiftExample
//
//  Created by Jeffrey Bakker on 2015-05-20.
//  Copyright (c) 2015 Jeffrey Bakker. All rights reserved.
//

import Foundation

class Marshmallow: NSObject, Topping
{
    func name() -> String
    {
        return "\(_stdlib_getDemangledTypeName(self))"
    }
}
